Assets/_Project/Runtime/_Scripts/GameManager/GameManager.cs

```diff
@@ -0,0 +1,135 @@
+#region
+using System;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Custom.Attributes;
+using UnityEngine.SceneManagement;
+using static GameManager;
+#endregion
+
+[DisallowMultipleComponent]
+public class GameManager : SingletonPersistent<GameManager>
+{
+    HighScoreManager highScoreManager;
+    ScoreManager scoreManager;
+
+    [Header("UI Elements")]
+    public GameObject mainMenuPanel;
+    public GameObject pausePanel;
+    public GameObject gameOverPanel;
+
+    public bool isGameOver = false;
+
+
+    #region Singleton
+    private static GameManager _instance;
+    public static GameManager Instance
+    {
+        get
+        {
+            if (_instance == null)
+            {
+                Debug.LogError("GameManager instance not found!");
+            }
+            return _instance;
+        }
+    }
+
+    void Awake()
+    {
+        if (_instance != null && _instance != this)
+        {
+            Destroy(gameObject);
+            Debug.LogError("Singleton already exist");
+            return;
+        }
+
+        _instance = this;
+        DontDestroyOnLoad(gameObject);
+    }
+    #endregion
+
+    public GameState state;
+
+    public enum GameState
+    {
+        Play,
+        Pause,
+        GameOver
+        // Add more game states if needed
+    }
+
+    void Start()
+    {
+        mainMenuPanel.SetActive(false);
+        gameOverPanel.SetActive(false);
+        pausePanel.SetActive(false);
+    }
+
+    void Update()
+    {
+        switch (state)
+        {
+            case GameState.Play:
+                if (Input.GetKeyDown(KeyCode.Escape))
```

Comment:
I'll convert all of this to the new InputSystem eventually

Reply:
👍